pkg/util/port/portlist/netstat_windows.go

```diff
@@ -0,0 +1,283 @@
+	// being logged at least.
+	GetModule() (string, error)
+}
+
+// See https://docs.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getextendedtcptable
+
+// TCP_TABLE_OWNER_MODULE_ALL means to include the PID and module. The table type
+// we get back from Windows depends on AF_INET vs AF_INET6:
+// MIB_TCPTABLE_OWNER_MODULE for v4 or MIB_TCP6TABLE_OWNER_MODULE for v6.
+const tcpTableOwnerModuleAll = 8
+
+// TCPIP_OWNER_MODULE_BASIC_INFO means to request "basic information" about the
+// owner module.
+const tcpipOwnerModuleBasicInfo = 0
+
+var (
+	iphlpapi                    = windows.NewLazySystemDLL("iphlpapi.dll")
+	getTCPTable                 = iphlpapi.NewProc("GetExtendedTcpTable")
+	getOwnerModuleFromTCPEntry  = iphlpapi.NewProc("GetOwnerModuleFromTcpEntry")
+	getOwnerModuleFromTCP6Entry = iphlpapi.NewProc("GetOwnerModuleFromTcp6Entry")
+	// TODO: GetExtendedUdpTable also? if/when needed.
+)
+
+type _MIB_TCPROW_OWNER_MODULE struct { //nolint:revive // Windows API type
+	state            uint32
+	localAddr        uint32
+	localPort        uint32
+	remoteAddr       uint32
+	remotePort       uint32
+	pid              uint32
+	createTimestamp  int64
+	owningModuleInfo [16]uint64
+}
+
+func ipport4(addr uint32, port uint16) netip.AddrPort {
+	if !cpu.IsBigEndian {
+		addr = bits.ReverseBytes32(addr)
+	}
+	return netip.AddrPortFrom(
+		netip.AddrFrom4([4]byte{byte(addr >> 24), byte(addr >> 16), byte(addr >> 8), byte(addr)}),
+		port)
+}
+
+func port(v *uint32) uint16 {
+	if !cpu.IsBigEndian {
+		return uint16(bits.ReverseBytes32(*v) >> 16)
+	}
+	return uint16(*v >> 16)
+}
+
+var states = []string{
+	"",
+	"CLOSED",
+	"LISTEN",
+	"SYN-SENT",
+	"SYN-RECEIVED",
+	"ESTABLISHED",
+	"FIN-WAIT-1",
+	"FIN-WAIT-2",
+	"CLOSE-WAIT",
+	"CLOSING",
+	"LAST-ACK",
+	"DELETE-TCB",
+}
+
+func state(v uint32) string {
+	if v < uint32(len(states)) {
+		return states[v]
+	}
+	return fmt.Sprintf("unknown-state-%d", v)
+}
+
+// See https://web.archive.org/web/20221219213143/https://learn.microsoft.com/en-us/windows/win32/api/iprtrmib/ns-iprtrmib-tcpip_owner_module_basic_info
+type _TCPIP_OWNER_MODULE_BASIC_INFO struct { //nolint:revive // Windows API type
+	moduleName *uint16
+	modulePath *uint16
+}
+
+type moduleInfoConstraint interface {
+	_MIB_TCPROW_OWNER_MODULE | _MIB_TCP6ROW_OWNER_MODULE
+}
+
+// See https://web.archive.org/web/20221219212442/https://learn.microsoft.com/en-us/windows/win32/api/tcpmib/ns-tcpmib-mib_tcp6row_owner_module
+type _MIB_TCP6ROW_OWNER_MODULE struct { //nolint:revive // Windows API type
+	localAddr        [16]byte
+	localScope       uint32
+	localPort        uint32
+	remoteAddr       [16]byte
+	remoteScope      uint32
+	remotePort       uint32
+	state            uint32
+	pid              uint32
+	createTimestamp  int64
+	owningModuleInfo [16]uint64
+}
+
+func moduleInfo[entryType moduleInfoConstraint](entry *entryType, proc *windows.LazyProc) (string, error) {
```

Comment:
I do not know how it would work in context of agent.exe process because by default it runs as non-administrator,  But a comment from https://learn.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getownermodulefromtcpentry 

> For computers running on Windows Vista or later, accessing the pModuleName and pModulePath members of the [TCPIP_OWNER_MODULE_BASIC_INFO](https://learn.microsoft.com/en-us/windows/desktop/api/iprtrmib/ns-iprtrmib-tcpip_owner_module_basic_info) structure is limited by user account control (UAC). If an application that calls this function is executed by a user logged on as a member of the Administrators group other than the built-in Administrator, this call will succeed but access to these members returns an empty string unless the application has been marked in the manifest file with a requestedExecutionLevel set to requireAdministrator. If the application on Windows Vista or later lacks this manifest file, a user logged on as a member of the Administrators group other than the built-in Administrator must then be executing the application in an enhanced shell as the built-in Administrator (RunAs administrator) for access to the protected pModuleName and pModulePath members to be allowed.

How did it work for you. It is not a new limitation, builtin netstat has the same problem

Reply:
Yeah I'm running into a problem because the diagnose command can tell that the APM and process agent ports are being utilized, but it can't tell what process is utilizing them. So we end up getting failures here:
```
24. --------------
  FAIL apm_config.debug.port
  Diagnosis: Required port 5012 is already used by an another process (PID=2204).

24. --------------
  FAIL apm_config.receiver_port
  Diagnosis: Required port 8126 is already used by an another process (PID=2204).

24. --------------
  FAIL process_config.expvar_port
  Diagnosis: Required port 6062 is already used by an another process (PID=3640).

24. --------------
  FAIL process_config.cmd_port
  Diagnosis: Required port 6162 is already used by an another process (PID=3640).
```
Another bug is that the list outputs every entry as 24 when it should be rising from 1-24. 

Do you have any ideas on how to check the process name from the process ID? Otherwise I might just add some verbiage to say "Check if process ID xxxx is an Agent component or something else."